src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/repository.py

```diff
@@ -42,47 +39,94 @@ def _bearer_auth_str(token):
+
+def _delete_data_from_registry(login_server, path, username, password, retry_times=3, retry_interval=5):
+    for i in range(0, retry_times):
+        errorMessage = None
+        try:
+            response = requests.delete(
+                'https://{}/{}'.format(login_server, path),
+                headers=_authorization_header(username, password)
+            )
+
+            if response.status_code == 200 or response.status_code == 202:
+                return
+            elif response.status_code == 401:
+                raise Unauthorized(response.text)
+            elif response.status_code == 404:
+                raise NotFound(response.text)
+            else:
+                raise CLIError(response.text)
+        except NotFound:
+            raise
+        except Unauthorized:
+            raise
+        except Exception as e:  # pylint: disable=broad-except
+            errorMessage = str(e)
+            logger.debug('Retrying %s with exception %s', i + 1, errorMessage)
+            time.sleep(retry_interval)
     else:
-        raise CLIError(response.text)
+        raise CLIError(errorMessage)
+
+
+def _get_manifest_digest(login_server, path, username, password, retry_times=3, retry_interval=5):
+    for i in range(0, retry_times):
+        errorMessage = None
+        try:
+            headers = _authorization_header(username, password)
+            headers.update(_manifest_v2_header())
+            response = requests.get(
+                'https://{}/{}'.format(login_server, path),
+                headers=headers
+            )
+
+            if response.status_code == 200 and response.headers and 'Docker-Content-Digest' in response.headers:
+                return response.headers['Docker-Content-Digest']
+            elif response.status_code == 401:
+                raise Unauthorized(response.text)
+            elif response.status_code == 404:
+                raise NotFound(response.text)
+            else:
+                raise CLIError(response.text)
+        except NotFound:
+            raise
+        except Unauthorized:
+            raise
+        except Exception as e:  # pylint: disable=broad-except
+            errorMessage = str(e)
+            logger.debug('Retrying %s with exception %s', i + 1, errorMessage)
+            time.sleep(retry_interval)
+    else:
+        raise CLIError(errorMessage)
 
 
 def _obtain_data_from_registry(login_server,
                                path,
                                username,
                                password,
-                               resultIndex=None,
+                               result_index,
                                retry_times=3,
-                               retry_interval=5):
-    registryEndpoint = 'https://' + login_server
+                               retry_interval=5,
+                               pagination=20):
+    print(login_server)
+    print(username)
+    print(password)
+    print(pagination)
```

Comment:
Wait ... what's this? Please don't print password.

Reply:
Missed the cleanup here :-)